ofrak_core/ofrak/core/ubifs.py

```diff
@@ -0,0 +1,204 @@
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File, Folder, FilesystemRoot, SpecialFileType
+from ofrak.core.binary import GenericBinary
+
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak_type.range import Range
+
+from ubireader.ubifs.defines import UBIFS_NODE_MAGIC
+
+from ubireader import ubi_io
+from ubireader.ubifs import ubifs as ubireader_ubifs
+from ubireader.ubifs.defines import PRINT_UBIFS_KEY_HASH, PRINT_UBIFS_COMPR
+from ubireader.utils import guess_leb_size
+
+LOGGER = logging.getLogger(__name__)
+
+MKFS_UBIFS_TOOL = ComponentExternalTool(
+    "mkfs.ubifs",
+    "http://www.linux-mtd.infradead.org/faq/ubifs.html",
+    install_check_arg="--help",
+    apt_package="mtd-utils",
+    brew_package=None,  # This isn't compatible with macos, but there may be an alternative tool to do the bidding.
+)
+
+
+@dataclass
+class SuperblockNode:
+    """
+    Each UBIFS image has a superblock which describe a large number of parameters regarding the filesystem. The
+    minimal set of parameters necessary to re-pack an UBIFS filesystem are stored here.
+    (see also: https://elixir.bootlin.com/linux/v6.1.7/source/fs/ubifs/ubifs.h#L1017).
+
+
+    :cvar max_leb_count: Maximum number / limit of Logical Erase Blocks
+    :cvar default_compr: Default compression algorithm
+    :cvar fanout: Fanout of the index tree (number of links per indexing node)
+    :cvar key_hash: Type of hash function used for keying direntries (typically 'r5' of reiserfs)
+    :cvar orph_lebs: Number of LEBs used for orphan area (orphans are inodes with no links; see https://elixir.bootlin.com/linux/v6.1.7/source/fs/ubifs/orphan.c#L13)
+    :cvar log_lebs: LEBs reserved for the journal (see the 'Journal' section in https://www.kernel.org/doc/Documentation/filesystems/ubifs-authentication.rst)
+    """
+
+    max_leb_count: int
+    default_compr: str  # PRINT_UBIFS_COMPR
+    fanout: int
+    key_hash: str  # PRINT_UBIFS_KEY_HASH
+    orph_lebs: int
+    log_lebs: int
+
+
+@dataclass
+class Ubifs(GenericBinary, FilesystemRoot):
+    """
+    UBIFS is a filesystem specially made to run on top of an UBI volume layer. UBIFS specifically provides
+    indexing, compression, encryption / authentication and some other filesystem-related features.
+
+    As part of an UBI image, re-packing an UBIFS image requires the 'min_io_size' and 'leb_size' properties that
+    are stored as part of the UBI header (https://elixir.bootlin.com/linux/v6.1.7/source/drivers/mtd/ubi/ubi.h#L441).
+
+    Each UBIFS image has a superblock which is encoded in OFRAK as a SuperblockNode.
+
+    Some documentation about UBIFS layout can also be found here:
+    https://www.kernel.org/doc/Documentation/filesystems/ubifs-authentication.rst
+    http://www.linux-mtd.infradead.org/doc/ubifs.html
+
+    :cvar min_io_size: Minimum number of bytes per I/O transaction (see http://www.linux-mtd.infradead.org/doc/ubi.html#L_min_io_unit)
```

Comment:
Have you tested that `cvar` renders correctly in the built docs? I know that `var` and `ivar` both work

I think technically these are not class variables -- it looks like class variables are specifically skipped by dataclass: https://docs.python.org/3/library/dataclasses.html#class-variables. 

Let's just change all of these to `:var ...`?